app/invocation/invocation_suggestion_card.tsx

```diff
@@ -70,6 +71,109 @@ const matchers: SuggestionMatcher[] = [
+        </>
+      ),
+      reason: (
+        <>
+          Shown because this build has the effective flag{flags.length === 1 ? "" : "s"}{" "}
+          {flags.map((flag, i) => (
+            <>
+              <span className="inline-code">
+                --{flag}={model.optionsMap.get(flag)}
+              </span>
+              {i < flags.length - 1 && " "}
+            </>
+          ))}
+        </>
+      ),
+    };
+  },
+  // Suggest using cache compression
+  ({ model }) => {
+    if (!capabilities.config.expandedSuggestionsEnabled) return null;
+
+    if (model.optionsMap.get("experimental_remote_cache_compression")) return null;
+    if (!model.optionsMap.get("remote_cache") && !model.optionsMap.get("remote_executor")) return null;
+    const version = model.started?.buildToolVersion;
+    if (!version) return null;
+    const segments = version.split(".").map(Number);
+    // Bazel pre-v5 doesn't support compression.
+    if (!segments[0] || segments[0] < 5) return null;
+
+    return {
+      level: SuggestionLevel.INFO,
+      message: (
+        <>
+          Consider adding the Bazel flag <span className="inline-code">--experimental_remote_cache_compression</span> to
+          improve remote cache throughput.
+        </>
+      ),
+      reason: <>Shown because this build is cache-enabled but compression is not enabled.</>,
+    };
+  },
+  // Suggest using --jobs
+  ({ model }) => {
+    if (!capabilities.config.expandedSuggestionsEnabled) return null;
+
+    if (!model.optionsMap.get("remote_executor")) return null;
+    if (model.optionsMap.get("jobs")) return null;
+
+    return {
+      level: SuggestionLevel.INFO,
+      message: (
+        <>
+          Configuring a jobs count will allow more actions to execute in parallel, which can significantly improve
+          remote execution performance. We recommend starting with <span className="inline-code">--jobs=50</span> and
+          working your way up.
+        </>
+      ),
+      reason: <>Shown because this build has remote execution enabled, but a jobs count is not configured.</>,
+    };
+  },
+  // Suggest using remote_download_minimal
+  ({ model }) => {
+    if (!capabilities.config.expandedSuggestionsEnabled) return null;
+
+    if (!model.optionsMap.get("remote_cache") && !model.optionsMap.get("remote_executor")) return null;
+    if (model.optionsMap.get("remote_download_outputs")) return null;
+
+    return {
+      level: SuggestionLevel.INFO,
+      message: (
+        <>
+          Adding the Bazel flag <span className="inline-code">--remote_download_minimal</span> or{" "}
+          <span className="inline-code">--remote_download_toplevel</span> can significantly improve performance for
+          builds with a large number of intermediate results.
+        </>
+      ),
+      reason: (
+        <>
+          Shown because this build is cache-enabled and the effective flag{" "}
```

Comment:
Each of these starts with "Shown because". Seems a bit redundant.

Reply:
do you mean redundant from a code perspective ("DRY") or redundant from the user's perspective?

If the latter, I think the redundancy serves to reinforce that the bottom "footnote" is always just explaining the technical reason that the suggestion is displayed, and that it can be ignored when skimming over these suggestions unless you are confused why this suggestion is being displayed:

![image](https://user-images.githubusercontent.com/2414826/169095852-6beda114-03ab-4e6b-ad2c-02334410844f.png)

Reply:
From a user's perspective. Seeing it in your screenshot though, I think "Reason: " would read better than "Shown because".